tests/S3/StreamWrapperTest.php

```diff
@@ -409,6 +409,16 @@ public function testCanDeleteNestedFolderWithRmDir()
 
     /**
      * @expectedException \PHPUnit_Framework_Error_Warning
+     * @expectedExceptionMessage rename(): Cannot rename a file across wrapper types
+     */
+    public function testRenameEnsuresProtocolsMatch()
```

Comment:
Actually, I'm beginning to like this now since if we ever forget, or behavior in PHP changes, we'll probably need to modify `rename` accordingly.